main.cpp

```diff
@@ -121,385 +163,563 @@ struct CarWash
-5 properties:
-    1) Attack time in ms
-    2) Decay time in ms
-    3) Sustain level in dB
-    4) Release time in ms
-    5) Amount of MIDI velocity level as a percentage
-3 things it can do:
-    1) control amplitude level of a sample over time
-    2) control filter cutoff frequency over time 
-    3) control pitch of a sample over time
- */
+void Filter::CircuitTypeEmulator::internallyFeedback(float feedbackValue)
+{
+    if(feedbackValue > 0.7f)
+    {
+        selfOscAmount += feedbackValue;
+    }
+}
+
+void Filter::CircuitTypeEmulator::clipAudio(std::string clipType)
+{
+    clipType = "Soft";
+}
+
+void Filter::CircuitTypeEmulator::manipulatePhase(bool isLinearPhase)
+{
+    if(isLinearPhase == true)
+    {
+        slope = 24;
+    }
+    else 
+    {
+        slope = 12;
+    }
+}
+
+void Filter::freqContentAffected(double lowFreq, double highFreq, double freqCutoffChange)
+{
+    lowFreq = 500;
+    highFreq = 4000;
+    freqCutoffChange = 2000;
+    freqCutoff += freqCutoffChange;
+}
+
+void Filter::distFreqContent(bool distTurnedUp, float distortionAmount, int driveAmount)
+{
+    distTurnedUp = true;
+    distortionAmount = 1.1f;
+    driveAmount = 3;
+}
+
+void Filter::midiNoteToCutoff(int midiNoteValue, int keyTrackerAmount = 3)
+{
+    midiNoteValue = 45;
+    m_keyTrackerAmount += keyTrackerAmount;
+}
+    
+// UDT 8 // complete
+
 struct Envelope
 {
-    //Attack time in ms
     float attTime = 0.1f;
-    //Decay time in ms
     float decTime = 0.5f;
-    //Sustain level in dB
     float susLevel = 0.0f;
-    //Release time in ms
     float relTime = 1.5f; 
-    //Amount of MIDI velocity level as a percentage
     int midiVelLevel = 75;
 
-    //control amplitude level of a sample over time
-    void controlAmpOverTime(double maxValueAmp, float sampleLength, bool iskeyPressed, bool iskeyReleased);
-    //control filter cutoff frequency over time 
-    void controlCutoffOverTime(double maxValueCutoff, double freqCutoff, bool iskeyPressed, bool iskeyReleased);
-    //control pitch of a sample over time
-    void controlPitchOverTime(double maxValuePitch, double minValue, bool iskeyPressed, bool iskeyReleased);
+    void controlAmpOverTime(bool ampEnvSelected, bool isKeyPressed, float maxValueAmp);
+    void controlCutoffOverTime(bool filterEnvSelected, bool isKeyPressed, double maxValueFilter);
+    void controlPitchOverTime(bool pitchEnvSelected, bool isKeyPressed, double maxValuePitch);
 };
 
-/*
-Thing 9) LFO
-5 properties:
-    1) Rate in Hz
-    2) Modulation amount as a percentage
-    3) Modulation destination 
-    4) Waveform type
-    5) Attack in ms
-3 things it can do:
-    1) modulate amplitude of sample 
-    2) modulate pitch of sample
-    3) modulate audio output between left and right channel
- */
+void Envelope::controlAmpOverTime(bool ampEnvSelected, bool isKeyPressed, float maxValueAmp)
+{
+    if(isKeyPressed == true && ampEnvSelected == true)
+    {
+        maxValueAmp = midiVelLevel / 100;
+    }
```

Comment:
You don't need to compare these booleans with `true`.  `if( isKeyPressed && ampEnvSelected )` will do the trick
- [ ] revise